os/pm/pm_register.c

```diff
@@ -83,19 +83,21 @@
  *
  ****************************************************************************/
 
-int pm_register(int domain_indx, FAR struct pm_callback_s *callbacks)
+int pm_register(FAR struct pm_callback_s *callbacks)
 {
 	int ret;
 
 	DEBUGASSERT(callbacks);
 
 	/* Add the new entry to the end of the list of registered callbacks */
 
-	ret = pm_lock(domain_indx);
-	if (ret == OK) {
-		sq_addlast(&callbacks->entry, &g_pmglobals.domain[domain_indx].registry);
+	ret = pm_lock();
+	if (ret == OK)
+	{
+		dq_addlast(&callbacks->entry, &g_pmglobals.registry);
+		pm_unlock();
 	}
-	pm_unlock(domain_indx);
+
```

Comment:
Is it okay not to unlock the pm when `ret` is not OK?

Reply:
when ret is not OK, it means it did not acquired the lock. Hence no unlock needed.